Content.Server/Fax/FaxSystem.cs

```diff
@@ -0,0 +1,390 @@
+        {
+            { DeviceNetworkConstants.Command, FaxConstants.FaxPingCommand }
+        };
+
+        if (component.Emagged)
+            payload.Add(FaxConstants.FaxSyndicateData, true);
+
+        _deviceNetworkSystem.QueuePacket(uid, null, payload);
+    }
+
+    /// <summary>
+    ///     Sends message to addressee if paper is set and a known fax is selected
+    ///     A timeout is set after sending
+    /// </summary>
+    public void Send(EntityUid uid, FaxMachineComponent? component = null)
+    {
+        if (!Resolve(uid, ref component))
+            return;
+
+        var sendEntity = component.PaperSlot.Item;
+        if (sendEntity == null)
+            return;
+
+        if (component.DestinationFaxAddress == null)
+            return;
+        
+        if (!component.KnownFaxes.TryGetValue(component.DestinationFaxAddress, out var faxName))
+            return;
+
+        if (!TryComp<PaperComponent>(sendEntity, out var paper))
+            return;
+
+        var payload = new NetworkPayload()
+        {
+            { DeviceNetworkConstants.Command, FaxConstants.FaxPrintCommand },
+            { FaxConstants.FaxContentData, paper.Content }
+        };
+        _deviceNetworkSystem.QueuePacket(uid, component.DestinationFaxAddress, payload);
+
+        component.SendTimeoutRemaining += component.SendTimeout;
+        
+        _audioSystem.PlayPvs(component.SendSound, uid);
+        
+        UpdateUserInterface(uid, component);
+    }
+
+    /// <summary>
+    ///     Accepts a new message and adds it to the queue to print
+    ///     If has parameter "notifyAdmins" also output a special message to admin chat.
+    /// </summary>
+    public void Receive(EntityUid uid, string content, string? fromAddress, FaxMachineComponent? component = null)
+    {
+        if (!Resolve(uid, ref component))
+            return;
+
+        var faxName = Loc.GetString("fax-machine-popup-source-unknown");
+        if (fromAddress != null && component.KnownFaxes.ContainsKey(fromAddress)) // If message received from unknown for fax address
+            faxName = component.KnownFaxes[fromAddress];
+
+        _popupSystem.PopupEntity(Loc.GetString("fax-machine-popup-received", ("from", faxName)), uid, Filter.Pvs(uid));
+        _appearanceSystem.SetData(uid, FaxMachineVisuals.VisualState, FaxMachineVisualState.Printing);
+
+        if (component.NotifyAdmins)
+            NotifyAdmins(faxName);
+
+        component.PrintingQueue.Enqueue(content);
+    }
+
+    private void SpawnPaperFromQueue(EntityUid uid, FaxMachineComponent? component = null)
+    {
+        if (!Resolve(uid, ref component) || component.PrintingQueue.Count == 0)
+            return;
+
+        var content = component.PrintingQueue.Dequeue();
+        var printed = EntityManager.SpawnEntity("Paper", Transform(uid).Coordinates);
+        if (TryComp<PaperComponent>(printed, out var paper))
+            _paperSystem.SetContent(printed, content);
```

Comment:
IMO the printed entity should have a name indicating the source, e.g. "message from centcom", so you don't have to be in view of the pop-up to identify the source.

Or maybe just make the document name part of the payload?

Reply:
Now pass name in payload